pages/categories/[id].js

```diff
@@ -139,28 +142,26 @@ const SubcategoryPage = ({ fetcher, favorites, onToggleFavorite }) => {
 
       <main>
         <CardWrapper>
-
-          
-       {filteredProviders.length > 0 ? (
+          {filteredProviders.length > 0 ? (
             filteredProviders.map((provider) => (
               <Card key={provider._id}>
-              {session && (
-
-                <FavoriteButton
-                  onClick={() => onToggleFavorite(provider._id)}
-                  isFavorite={favorites.includes(provider._id)}
+                {session && session.user.email !== provider.author && (
+                  <FavoriteButton
+                    onClick={() => onToggleFavorite(provider._id)}
+                    isFavorite={favorites.includes(provider._id)}
+                  />
+                )}
+
+                <ServiceProvider
+                  key={provider._id}
+                  card={provider}
+                  isOnUserPage={true}
                 />
```

Comment:
```suggestion
                <ServiceProvider
                  key={provider._id}
                  card={provider}
                  isOnUserPage
                />
```

But this confuses me, this is on the categories route. Is this set up as intended?